properties/base.py

```diff
@@ -182,14 +182,18 @@ def _notify(self, change):
             listener.func(self, change)
 
     def _set(self, name, value):
-        out = self._notify(dict(name=name, value=value, mode='validate'))
-        if out is not None:
-            value = out
-        if value is basic.undefined and name in self._backend:
+        change = dict(name=name, value=value, mode='validate')
+        self._notify(change)
+        if change['name'] != name:
+            warn('Specified Property for assignment changed during '
```

Comment:
The validating observers are now able to coerce `value` in the `change` dictionary - that's the expected behaviour. There's no reason they can't change `name` in the `change` dictionary as well, but that undermines the property that is getting changed. Maybe it would be better to be stricter and throw an error? But certainly not expected behaviour.

Hmm, yeah, I think I might throw an error rather than warn - the way it is now, if `name` changes, a different property will be notified after the change than before. Ugly. Or maybe we should just trust that no one messes with `name` and do nothing...